internal/backends/python/pypi_map.go

```diff
@@ -0,0 +1,116 @@
+/*
+pypi_map is a convinience wrapper around the pypi_map.sqlite read-only database, which is
+generated by gen_pypi_map.go using db_gen.go.
+*/
+
+package python
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"strings"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type PypiMap struct {
+	db *sql.DB
+}
+
+func NewPypiMap() (*PypiMap, error) {
+	dbFilePath := os.Getenv("PYPI_MAP_DB")
+	if dbFilePath == "" {
+		return nil, fmt.Errorf("PYPI_MAP_DB not set. Set it to the path of the sqlite database.")
+	}
+	// Open db in read-only mode so it doesn't try to make any modifications
+	// which would cause it to copy on write in our layered fs
+	db, err := sql.Open("sqlite3", dbFilePath+"?mode=ro")
```

Comment:
Immutable should be faster and safe here because we are storing this int he Nix store, which won't change out from under us. https://www.sqlite.org/uri.html#recognized_query_parameters

```suggestion
	db, err := sql.Open("sqlite3", dbFilePath+"?immutable=1")
```

[Cribbed from the Layered store work ](https://github.com/NixOS/nix/pull/8356/files#diff-766e649c52f1b2f334ad2788aa2230b88b0a7cdd2799529c6125462960b145e0R508)

Reply:
We're definitely going to want to do this. I just tried it and it is faster (TM).

Reply:
Before on my computer:
```
[nix-shell:~/p/replit/upm/play]$ time ../result/bin/upm lock

real  0m0.081s
user  0m0.086s
sys 0m0.037s

[nix-shell:~/p/replit/upm/play]$ time ../result/bin/upm guess -fa
--> python3 /run/user/1000/upm3791543655/bare-imports.py '<secret sauce>'
flask
pyyaml
requests

real  0m0.124s
user  0m0.108s
sys 0m0.044s
```
After:

```
[nix-shell:~/p/replit/upm/play]$ time ../result/bin/upm lock

real  0m0.014s
user  0m0.006s
sys 0m0.009s

[nix-shell:~/p/replit/upm/play]$ time ../result/bin/upm guess -fa
--> python3 /run/user/1000/upm2231193298/bare-imports.py '<secret sauce>'
flask
pyyaml
requests

real  0m0.041s
user  0m0.030s
sys 0m0.013s
```

With immutable:

```
[nix-shell:~/p/replit/upm/play]$ time ../result/bin/upm lock

real  0m0.006s
user  0m0.002s
sys 0m0.005s

[nix-shell:~/p/replit/upm/play]$ time ../result/bin/upm guess -fa
--> python3 /run/user/1000/upm2868831452/bare-imports.py '<secret sauce>'
flask
pyyaml
requests

real  0m0.028s
user  0m0.023s
sys 0m0.007s
```